packages/mom5/package.py

```diff
@@ -5,114 +5,184 @@
+    maintainers("dougiesquire", "harshula", "penguian")
 
+    # https://github.com/ACCESS-NRI/MOM5#LGPL-3.0-1-ov-file
+    license("LGPL-3.0-only", checked_by="dougiesquire")
+
+    version("mom_solo", branch="master")
+    version("mom_sis", branch="master")
     version("access-om2", branch="master", preferred=True)
     version("legacy-access-om2-bgc", branch="master")
     version("access-esm1.5", branch="access-esm1.5")
     version("access-esm1.6", branch="master")
 
-    variant("restart_repro", default=True, description="Reproducible restart build.")
-
-    with when("@access-esm1.6,access-om2,legacy-access-om2-bgc"):
-        variant("deterministic",
-                default=False,
-                description="Deterministic build.")
-        variant("optimisation_report",
-                default=False,
-                description="Generate optimisation reports.")
+    # NOTE: @mom matches both mom_solo and mom_sis
+    build_system(
+        conditional("makefile", when="@access-om2,legacy-access-om2-bgc,access-esm1.5"),
+        conditional("cmake", when="@mom,access-om2,legacy-access-om2-bgc,access-esm1.6"),
+        default="cmake",
+    )
+
+    with when("build_system=cmake"):
+        variant("build_type", default="RelWithDebInfo",
+            description="CMake build type",
+            values=("Debug", "Release", "RelWithDebInfo")
+        )
+        variant("deterministic", default=False, description="Deterministic build")
+
+    with when("build_system=makefile"):
+        variant("restart_repro", default=True, description="Reproducible restart build.")
+        variant(
+            "deterministic",
+            default=False,
+            description="Deterministic build",
+            when="@access-om2,legacy-access-om2-bgc"
+        )
+        variant(
+            "optimisation_report",
+            default=False,
+            description="Generate optimisation reports",
+            when="@access-om2,legacy-access-om2-bgc"
+        )
 
-    with when("@access-om2,legacy-access-om2-bgc"):
+    with when("@mom,access-om2,legacy-access-om2-bgc,access-esm1.6"):
         depends_on("netcdf-c@4.7.4:")
         depends_on("netcdf-fortran@4.5.2:")
         # Depend on virtual package "mpi".
         depends_on("mpi")
-        depends_on("datetime-fortran")
+
+    with when("@access-om2,legacy-access-om2-bgc"):
         depends_on("oasis3-mct+deterministic", when="+deterministic")
         depends_on("oasis3-mct~deterministic", when="~deterministic")
+
+    with when("@access-esm1.6"):
+        depends_on("oasis3-mct@access-esm1.5+deterministic", when="+deterministic")
+        depends_on("oasis3-mct@access-esm1.5~deterministic", when="~deterministic")
+
+    with when("@access-om2,legacy-access-om2-bgc"):
+        depends_on("datetime-fortran")
         depends_on("libaccessom2+deterministic", when="+deterministic")
         depends_on("libaccessom2~deterministic", when="~deterministic")
```

Comment:
```suggestion
    with when("@access-om2,legacy-access-om2-bgc"):
        depends_on("datetime-fortran")
        depends_on("oasis3-mct+deterministic", when="+deterministic")
        depends_on("oasis3-mct~deterministic", when="~deterministic")
        depends_on("libaccessom2+deterministic", when="+deterministic")
        depends_on("libaccessom2~deterministic", when="~deterministic")

    with when("@access-esm1.6"):
        depends_on("oasis3-mct@access-esm1.5+deterministic", when="+deterministic")
        depends_on("oasis3-mct@access-esm1.5~deterministic", when="~deterministic")
```

Reply:
Oops - ta

Reply:
Since the changes in the last commit, there's an opportunity to consolidate the conditionals above.